### DISCUSSION TITLE
1.5.0

### DISCUSSION THREAD

**Discussion by ilyhalight** [ilyhalight/voice-over-translation#459] [Обновления]
## Список изменений
> [!WARNING]
> На 23.12.2023 cloudflare workers не работают с российскими айпи адресами. В связи с этим перевод стримов не будет работать без VPN.

- Добавлена эксперементальная поддержка трансляций (На 23.12.2023 не работает в РФ из-за ограничений на российский айпи от cloudflare worker (позже исправлю). С задержкой как повезет, но в большинстве случаев довольно терпимо +-0-2 секунды от оригинала, но иногда бывает отставание до 10 секунд, но перезапуск перевода помогает уменьшить задержку)
- Добавлен HLS.js
- Добавлена поддержка Yandex Disk (только публичные ссылки, например: `https://disk.yandex.ru/s/FILE_ID`)
- Добавлена поддержка Trovo (клипы или уже завершенные трансляции)
- Добавлена частичная поддержка Dailymotion ([34#issuecomment-1533772639](https://github.com/ilyhalight/voice-over-translation/issues/34#issuecomment-1533772639))

  > Перед выполнением перевода видео перезагрузите страницу, иначе вы рискуете перевести одно из старых просмотренных вами видео. Я не нашел способа как это исправить.

- Добавлена поддержка Peertube. Были добавлены 9 крупных сайтов, хостящих Peertube (libre.video не поддерживается - не просите):
  - tube.shanti.cafe
  - bee-tube.fr
  - video.sadmin.io
  - dalek.zone
  - review.peertube.biz
  - peervideo.club
  - tube.la-dina.net
  - peertube.tmp.rcp.tf
- Добавлена поддержка EPorner (#409)
- Добавлена поддержка YouTube Kids
- Добавлена поддержка rumble.com (#94, #123)
- Добавлена поддержка YT Shorts
- Для VK, Piped, Invidious, Bitchute, Rumble, Peertube, DailyMotion, Trovo, и Yandex Disk стандартный язык перевода изменен на "Auto"
- Добавлен слайдер регулировки автоматически устанавливаемой громкости видео при включение перевода (#418)
- Добавлена возможность установки своего прокси-домена без пересборки расширения
- Добавлена опция включающая перевод ошибок возвращаемых с сервера. Работает на базе [Яндекс переводчика](https://github.com/FOSWLY/translate-backend) (#429, #309)
- Добавлен выбора сервиса для определения языка видео:
  - [Yandex](https://github.com/FOSWLY/translate-backend) (должен определять точнее, установлен по умолчанию)
  - Rust-server (тот, что использовался раньше)
- Улучшено определение языка при отсутвии субтитров на YouTube (убрано использование имени автора каанала) от @SashaXser в #355
- Добавлена возможность загрузки субтитров от Яндекса в формате json (#352, #335)
- Добавлен таймаут в 3 секунды, если детект сервер не доступен
- Добавлен выбор языка для опции "Не переводить с родного языка" (если не трогать, то за основу берется язык браузера)
- Добавлена возможность проигрывания нескольких переводов одновремено
- Добавлена возможность кастомизации цветов расширения через Stylish, Stulus и другие подобные расширения

  Пример изменения цветов меню на зеленый:
  ```css

  /* ==UserStyle==
  @name         VOT-GREEN
  @version      16.09.2023
  @namespace    vot-green
  @description  LLL
  @author       Toil
  @license      No License
  ==/UserStyle== */

  :root {
    --vot-primary-rgb: 201, 245, 180 ;
  }

  .vot-segmented-button[data-status="success"] .vot-translate-button {
    color: rgb(201, 245, 180);
    fill: rgb(201, 245, 180);
  }
  ```

- Добавлено версионирование переводов
- Добавлена возможность включить кнопку PiP для браузеров в которых она поддерживается (хромиум)
- Добавлено немного информации в меню
- Добавлена возможность смены языка меню
- Исправлен небольшой баг с "Не переводить с родного языка" (#414)
- Фикс Coursera (#444)
- В список языков меню добавлены все доступные языки
- Добавлены доп. проверки на существование методов для YouTube player
- Исправлена ошибка из-за которой в Greasymonkey не работала установка громкости
- Фикс эффекта "перетаскивания" меню для всех браузеров (#417)
- Фикс "отложенной загрузки видео" для YouTube (#387, #368, #277. Возможно #378)
- Добавлен хук pre-commit для автоформатирования при добавление в git
- Игнорирование папок перенесено в отдельный файл `.eslintignore`
- Задан явный конфиг для prettier (нужен для нормальной работы форматирования в редакторе)
- Теперь, по умолчанию значение опции "слайдер громкости видео" было изменено на включен
- Два разных хранилища IndexedDB и LocalStorage были заменены на одно - GM_Storage (LocalStorage используется, если не доступен GM Storage)
- Фикс updateURL, downloadURL и namespace для минифицированных версий и бета версий (#385)
- Включена подгрузка бета локализаций в бета версиях
- Все языки, перевод на которые может быть недоступен, были помечены с помощью `❌`
- Теперь, если сервер определения языка текста не отвечает будет установлен английский язык (#361) от @SashaXser в #366
- Убрана проверка на премьеру для YouTube от @SashaXser в #366
- Значения lectureData, videoURL и subtitlesURL, теперь, показываются и без дебаг режима (udemy, coursera)
- В полезные ссылки в README добавлены пометки `vot-cli` и `cloudflare`, дабы число вопросов с ними сократилось
- Сайты, с которыми могут возникать ошибки CSP, были помечены в README как требующие доп. действий, дабы число вопросов с ними сократилось
- Теперь, в режиме разработки расширение имеет префикс "[DEBUG]"
- Метод getSubtitlesFileURL, теперь, имеет одинаковый порядок аргументов в Udemy и Coursera
- Изменена логика получения ссылки на видео для Udemy и Coursera

  Теперь, ссылка на видео ищется не только в формате .mp4, но еще и в .webm.
  Добавлена проверка на то, является ли ссылка на видео блобом (если ссылка получена через плеер)

- Изменена логика получения ссылки на субтитры для Udemy и Coursera

  Теперь, проверка, изначально, идет по языку курса.
  Если субтитры по языку курса не были найдены, то идет проверка по языку на который выполняется перевод.
  Если и после этого субтитры не были найдены, то берутся первые субтитры из массива с субтитрами.
- Переписан UI расширения
- Переписана логика загрузки
- Кеш переводов, теперь, очищается при сбросе настроек
- Стандартный шрифт изменен на Roboto

## What's Changed
* VideoObserver, New UI and stuff by @MrSoczekXD in https://github.com/ilyhalight/voice-over-translation/pull/294
* Протестирована поддержка violentmonkey by @Spp595 in https://github.com/ilyhalight/voice-over-translation/pull/338
* Добавить языковые значки by @szepeviktor in https://github.com/ilyhalight/voice-over-translation/pull/342
* Доработка определение языка при отсутвии субтитров  by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/355
* Удаление использования isPremiere by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/366
* Update headers.json by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/456
* 1.5.0 by @ilyhalight in https://github.com/ilyhalight/voice-over-translation/pull/457
* 1.5.0 update by @ilyhalight in https://github.com/ilyhalight/voice-over-translation/pull/458

## New Contributors
* @Spp595 made their first contribution in https://github.com/ilyhalight/voice-over-translation/pull/338
* @szepeviktor made their first contribution in https://github.com/ilyhalight/voice-over-translation/pull/342

**Full Changelog**: https://github.com/ilyhalight/voice-over-translation/compare/1.4.1...1.5.0

<hr /><em>This discussion was created from the release <a href='https://github.com/ilyhalight/voice-over-translation/releases/tag/1.5.0'>1.5.0</a>.</em>

**Comment by NullVerdict** [comment]
Новый сервис для определения языка видео кажется умер...
![изображение](https://github.com/ilyhalight/voice-over-translation/assets/24498484/2bd083eb-5d2b-432e-9dfb-20028a849095)

**Reply by ilyhalight** [reply]
Если до вечера не встанет, то потом перенесу на другой хост

**Comment by Valun4e** [comment]
Ребята, вы на Ютуб заходили, со времени обновы скрипта? Если нет, то пожалуйста, зайдите и посмотрите(послушайте) на то, что там натворила версия 1.5.0 
Я не буду описывать все её "прелести", но у меня лишь один вопрос: есть ли (у меня) возможность вернуться(желательно, без особо заковыристых танцев с бубном) к прекрасной и беспроблемной версии 1.4.1? Если нет, то я больше не буду вас беспокоить своим нытьем и так и быть: установлю этот мерзкий, тормознутый и громоздкий Яндекс-браузер, ради видосов на английском языке. Ответьте, пожалуйста!

**Reply by NullVerdict** [reply]
Зайди в релизы и скачай нужную версию. Лично у меня проблем с 1.5.0 не было и нету.

**Reply by Valun4e** [reply]
А как? Нахожу 1.4.1, нажимаю на "go to file" и на "установите скрипт"(тамперманкей уже есть). Но мне всё равно предлагают скачать версию 1.5.0

**Reply by ilyhalight** [reply]
> А как? Нахожу 1.4.1, нажимаю на "go to file" и на "установите скрипт"(тамперманкей уже есть). Но мне всё равно предлагают скачать версию 1.5.0

Ищешь 1.4.1 в [версиях](https://github.com/ilyhalight/voice-over-translation/releases), раскрываешь список ассетов и выбираешь нужную версию скрипта (обычная/клаудфлаер/мин. обычная/мин. клаудфлаер) и нажимаешь на откатить. В настройках тамперманки запрещаешь автообновления
![изображение](https://github.com/ilyhalight/voice-over-translation/assets/62353659/eb603e87-0357-406a-a52f-74372fc80df2)
![изображение](https://github.com/ilyhalight/voice-over-translation/assets/62353659/731c64be-a1c2-4b84-ac94-30d15bc112ab)
![изображение](https://github.com/ilyhalight/voice-over-translation/assets/62353659/0c94e636-4d78-4097-907a-61c203570516)
![изображение](https://github.com/ilyhalight/voice-over-translation/assets/62353659/165e742a-45c5-498a-bac7-4c19f0e68b96)

**Comment by Valun4e** [comment]
Тьфу-тьфу чтоб не сглазить, но проблему кажись решил, откатом. И вуаля: всё стало как прежде! Боже, как прекрасна версия 1.4.1, она просто идеальна! Спасибо ребята за ваши труды и всех вам благ!

**Reply by Prottok** [reply]
Да..новая версия и в прям глючная, у меня окно с переводом и вовсе не отображается в CentBrowser... Хорошо хоть 1.4.1 работает как часы.

**Reply by ilyhalight** [reply]
> Да..новая версия и в прям глючная, у меня окно с переводом и вовсе не отображается в CentBrowser... Хорошо хоть 1.4.1 работает как часы.

Расширение не тестируется в старых браузерах, поэтому никто не гарантирует работу в Chromium<100 и FF<100 версии. Проблему из-за которой кнопка не появлялась исправил в 1.5.0.1

**Comment by Firsto** [comment]
Новый дизайн супер вообще :)

**Comment by NullVerdict** [comment]
Надо будет в будущем додумать одну идею, хочется "исправлять" субтитры Яндекса на что-то вменяемое. 
Потому-что Яндекс Переводчик переводит полную ахинею, а система контекста просто убогая. 
Даже если прогнать субтитры почти через любую языковую модель, то перевод будет в разы лучше.
Например: Скачиваются субтитры от Яндекса или видео (желательно не автоматические), прогоняются через какой нибудь Gemini Pro или другую языковую модель, после этого переведенные субтитры отправляются Яндексу на озвучку.

![План](https://github.com/ilyhalight/voice-over-translation/assets/24498484/78aa73ff-b979-419a-b087-c22f7c9cc1da)

**Comment by Janikaest** [comment]
Добрый вечер. Вначале хочу сказать большое спасибо за расширение, оно великолепно!
А есть ли возможность изменить голос на женский? Все перекопала, не могу найти. В Яндексе вроде была такая возможность.

**Comment by NullVerdict** [comment]
> А есть ли возможность изменить голос на женский?

Нельзя, голос определяется Яндексом автоматически. 
> В Яндексе вроде была такая возможность.

Никогда не было.
